libraries/Microsoft.Bot.Builder.Core/BotContext.cs

```diff
@@ -39,78 +39,97 @@ public BotContext(BotAdapter bot, ConversationReference conversationReference)
             _adapter = bot ?? throw new ArgumentNullException(nameof(bot));
             _conversationReference = conversationReference ?? throw new ArgumentNullException(nameof(conversationReference));
         }
-
-        //public async Task SendActivity(IBotContext context, IList<IActivity> activities)
-        //{
-        //    await _bot.SendActivity(context, activities).ConfigureAwait(false);
-        //}
-
-        public IActivity Request => _request;
-
         public BotAdapter Adapter => _adapter;
 
-        public IList<IActivity> Responses { get => _responses; set => this._responses = value; }
+        public Activity Request => _request;
 
-        public IStorage Storage { get; set; }
+        public IList<Activity> Responses { get => _responses; set => this._responses = value; }
 
-        public Intent TopIntent { get; set; }
-
-        public TemplateManager TemplateManager { get; set; }
+        public ConversationReference ConversationReference { get => _conversationReference; }
 
-        public bool IfIntent(string intentName)
+        public IBotContext Reply(string text, string speak = null)
         {
-            if (string.IsNullOrWhiteSpace(intentName))
-                throw new ArgumentNullException(nameof(intentName));
-
-            if (this.TopIntent != null)
+            var reply = this.ConversationReference.GetPostToUserMessage();
+            reply.Text = text;
+            if (!string.IsNullOrWhiteSpace(speak))
             {
-                if (TopIntent.Name == intentName)
-                {
-                    return true;
-                }
+                // Developer included SSML to attach to the message.
+                reply.Speak = speak;
             }
+            this.Responses.Add(reply);
+            return this;
+        }
 
-            return false;
+        public IBotContext Reply(IActivity activity)
+        {
+            BotAssert.ActivityNotNull(activity);
+            this.Responses.Add((Activity)activity);
+            return this;
         }
-        public bool IfIntent(Regex expression)
+
+        public void Set(string serviceId, object service)
```

Comment:
>service [](start = 49, length = 7)

Why can I only store services on the context? What if I just want to store a flag or a string? My implementation makes this just a generic map/dictionary that can store any key/value thing you want.

Reply:
You can, but I am using naming to push peoples thought pattern around what goes in there, even though you can put anything

---
In reply to: [170474805](https://github.com/Microsoft/botbuilder-dotnet/pull/193#discussion_r170474805) [](ancestors = 170474805)

Reply:
I just that people will think that's all we expect you to store on the context object.